cmd/apiProviderDownload_test.go

```diff
@@ -0,0 +1,79 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type apiProviderDownloadTestUtilsBundle struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func apiProviderDownloadMockUtilsBundle() *apiProviderDownloadTestUtilsBundle {
+	utilsBundle := apiProviderDownloadTestUtilsBundle{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return &utilsBundle
+}
+
+//Successful API Provider download cases
+func TestApiProviderDownloadSuccess(t *testing.T) {
+	t.Parallel()
+	t.Run("Successful Download of API Provider", func(t *testing.T) {
+		apiServiceKey := `{
+			"oauth": {
+				"url": "https://demo",
+				"clientid": "demouser",
+				"clientsecret": "******",
+				"tokenurl": "https://demo/oauth/token"
+			}
+		}`
+
+		config := apiProviderDownloadOptions{
+			APIServiceKey:   apiServiceKey,
+			APIProviderName: "provider1",
+		}
+		// test
+		httpClient := httpMockCpis{CPIFunction: "APIProviderDownload", ResponseBody: ``, TestType: "Positive"}
+		utilsMock := apiProviderDownloadMockUtilsBundle()
+		ioResp := runApiProviderDownload(&config, nil, &httpClient, utilsMock)
```

Comment:
If something is actually downloaded you don't care about, right? IMO, as the step is called xxxDownload a test that checks whether something was downloaded, i.e. written to a file, or not would make sense. What do you think?

Reply:
I have added an additional check for asserting file write. Line #46.

Reply:
Also, changed the variable ioresp to err, as the fileutil returns only error. Since I started with my own file util implementation, there were some inconsistencies.